contracts/ReputationMiningCycleRespond.sol

```diff
@@ -244,34 +237,29 @@ contract ReputationMiningCycleRespond is ReputationMiningCycleStorage, PatriciaT
   }
 
   function proveBeforeReputationValue(
-    uint256[11] u,
+    uint256[19] u,
     bytes _reputationKey,
     bytes32[] reputationSiblings,
-    bytes agreeStateReputationValue,
     bytes32[] agreeStateSiblings
   ) internal
   {
     bytes32 jrh = disputeRounds[u[U_ROUND]][u[U_IDX]].jrh;
     // We binary searched to the first disagreement, so the last agreement is the one before.
     uint256 lastAgreeIdx = disputeRounds[u[U_ROUND]][u[U_IDX]].lowerBound - 1;
-    uint256 reputationValue;
-    assembly {
-        reputationValue := mload(add(agreeStateReputationValue, 32))
-    }
+
+    bytes memory agreeStateReputationValue = concatenateToBytes64(u[U_AGREE_STATE_REPUTATION_VALUE], u[U_AGREE_STATE_REPUTATION_UID]);
 
     bytes32 reputationRootHash = getImpliedRoot(_reputationKey, agreeStateReputationValue, u[U_REPUTATION_BRANCH_MASK], reputationSiblings);
-    bytes memory jhLeafValue = new bytes(64);
+    bytes memory jhLeafValue = concatenateToBytes64(uint256(reputationRootHash), u[U_AGREE_STATE_NNODES]);
     bytes memory lastAgreeIdxBytes = new bytes(32);
+
     assembly {
-      mstore(add(jhLeafValue, 0x20), reputationRootHash)
-      let x := mload(add(u, mul(32,3))) // 3 = U_AGREE_STATE_NNODES. Constants not supported by inline solidity
-      mstore(add(jhLeafValue, 0x40), x)
       mstore(add(lastAgreeIdxBytes, 0x20), lastAgreeIdx)
```

Comment:
We can abstract the 3 instances where we calculate `lastAgreeIdxBytes`

Reply:
There is another instance of `firstDisagreeIdxBytes` but it is only a single one so probably not worth abstracting. This will take assembly level code in this contract down to 4 which makes me happy